### PR TITLE
showing qbo errors in the UI

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          {{element.type}}
+        </td>
+      </ng-container>
+
+      <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
+      <tr mat-row *matRowDef="let row; columns: columnsToDisplay;"></tr>
+    </table>
+
+  </div>
+</div>
```
Comment (Sravanksk) [inline src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html:40]: new line

File: src/app/core/models/task.model.ts [hunk 2]
```diff
@@ -8,6 +9,7 @@ export type Task = {
   credit_card_purchase: number;
   // having any here is okay, didn't differentiate qbo errors and fyle errors
   detail: any;
+  quickbooks_errors: QuickbookError[];
```
Comment (Sravanksk) [inline src/app/core/models/task.model.ts:12]: `quickbooks_errors: QuickbooksError[];`

File: src/app/core/models/quickbooks-error.model.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* tslint:disable */
+
+
+export type QuickbookError = {
```
Comment (Sravanksk) [inline src/app/core/models/quickbooks-error.model.ts:4]: `QuickbooksError`

File: src/app/core/models/quickbooks-error.model.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/* tslint:disable */
+
```
Comment (ashwin1111) [inline src/app/core/models/quickbooks-error.model.ts:2]: rm

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<div class="internal-loader" fxLayoutAlign="center center" fxFlexFill *ngIf="isLoading">
+  <app-loader></app-loader>
+</div>
+
+<div *ngIf="!count && !isLoading">
+  <zero-state message="This expense group has no Quickbooks Online errors"></zero-state>
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html:6]: this tab should be shown only when there are qbo errors

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts:1]: rm

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  columnsToDisplay = ['shortDescription', 'longDescription', 'type'];
+
+  constructor(private taskService: TasksService, private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    const that = this;
+    that.workspaceId = +that.route.snapshot.parent.params.workspace_id;
+    that.expenseGroupId = +that.route.snapshot.parent.params.expense_group_id;
+    that.isLoading = true;
+    that.taskService.getTasksByExpenseGroupId(that.expenseGroupId).subscribe((res: Task[]) => {
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts:31]: can you fix the same you did in other app? getTasksByExpenseGroupId should return just 1 task, fix it off it api as well as app.

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html (hunk 5 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html:6]: if that is the case, then why do you need a zero state?

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html [hunk 8]
```diff
@@ -16,9 +16,12 @@
       <a mat-tab-link (click)="changeState('INFO')" [active]="state==='INFO'">
         Info
       </a>
-      <a mat-tab-link (click)="changeState('MAPPING_ERRORS')" [active]="state==='MAPPING_ERRORS'">
+      <a mat-tab-link *ngIf="showMappingErrors" (click)="changeState('MAPPING_ERRORS')" [active]="state==='MAPPING_ERRORS'">
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html:19]: `state === 'MAPPING_ERRORS'`

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html [hunk 9]
```diff
@@ -16,9 +16,12 @@
...
         Mapping Errors
       </a>
+      <a mat-tab-link *ngIf="showQuickbooksErrors" (click)="changeState('QUICKBOOKS_ERRORS')" [active]="state==='QUICKBOOKS_ERRORS'">
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html:22]: spacing b/w ===

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.ts [hunk 10]
```diff
@@ -90,6 +92,8 @@ export class ViewExpenseGroupComponent implements OnInit {
       that.expenseGroup = response[0];
       if (response[1].length > 0) {
         that.task = response[1][0];
+        that.showMappingErrors = that.task.detail ? true : false;
```
Comment (ashwin1111) [inline src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.ts:95]: We store success logs OR mapping errors in detail column. I assume that doesn't affect it, please test it out and confirm here.

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.ts (hunk 10 above)
Comment (NileshPant1999) [reply src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.ts:95]: yes it is not affecting that.

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html (hunk 5 above)
Comment (NileshPant1999) [reply src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html:6]: yes true can remove this

**Review by Sravanksk — APPROVED** [review]

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html (hunk 5 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.html:6]: please fix this

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts (hunk 6 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts:1]: fix this

File: src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts (hunk 7 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/group-quickbooks-error/group-quickbooks-error.component.ts:31]: fix

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html (hunk 8 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html:19]: fix

File: src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html (hunk 9 above)
Comment (ashwin1111) [reply src/app/qbo/expense-groups/view-expense-group/view-expense-group.component.html:22]: fix

**Review by ashwin1111 — APPROVED** [review]

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**